guides/plugins/knn.md

```diff
@@ -0,0 +1,117 @@
+- [k-NN Plugin](#k-nn-plugin)
+  - [Basic Approximate k-NN](#basic-approximate-k-nn)
+    - [Create an Index](#create-an-index)
+    - [Index Vectors](#index-vectors)
+    - [Search for Nearest Neighbors](#search-for-nearest-neighbors)
+  - [Approximate k-NN with a Boolean Filter](#approximate-k-nn-with-a-boolean-filter)
+  - [Approximate k-NN with a Lucene Filter](#approximate-k-nn-with-a-lucene-filter)
+
+# k-NN Plugin
+
+Short for k-nearest neighbors, the k-NN plugin enables users to search for the k-nearest neighbors to a query point across an index of vectors. See [documentation](https://opensearch.org/docs/latest/search-plugins/knn/index/) for more information.
+
+## Basic Approximate k-NN
+
+In the following example we create a 5-dimensional k-NN index with random data. You can find a synchronous version of this working sample in [samples/knn/knn-basics.py](../../samples/knn/knn-basics.py) and an asynchronous one in [samples/knn/knn-async-basics.py](../../samples/knn/knn-async-basics.py).
+
+```bash
+$ poetry run knn/knn-basics.py
+
+Searching for [0.61, 0.05, 0.16, 0.75, 0.49] ...
+{'_index': 'my-index', '_id': '3', '_score': 0.9252405, '_source': {'values': [0.64, 0.3, 0.27, 0.68, 0.51]}}
+{'_index': 'my-index', '_id': '4', '_score': 0.802375, '_source': {'values': [0.49, 0.39, 0.21, 0.42, 0.42]}}
+{'_index': 'my-index', '_id': '8', '_score': 0.7826564, '_source': {'values': [0.33, 0.33, 0.42, 0.97, 0.56]}}
+```
+
+### Create an Index
+
+```python
+dimensions = 5
+client.indices.create(index_name, 
+    body={
+        "settings":{
+            "index.knn": True
+        },
+        "mappings":{
+            "properties": {
+                "values": {
+                    "type": "knn_vector", 
+                    "dimension": dimensions
+                },
+            }
+        }
+    }
+)
+```
+
+### Index Vectors
+
+Create 10 random vectors and insert them using the bulk API.
+
+```python
+vectors = []
+for i in range(10):
+    vec = []
+    for j in range(dimensions): 
+        vec.append(round(random.uniform(0, 1), 2)) 
+  
+    vectors.append({
+        "_index": index_name,
+        "_id": i,
+        "values": vec,
+    })
+
+helpers.bulk(client, vectors)
+
+client.indices.refresh(index=index_name)
+```
+
+### Search for Nearest Neighbors
+
+Create a random vector of the same size and search for its nearest neighbors.
+
+```python
+vec = []
+for j in range(dimensions): 
+    vec.append(round(random.uniform(0, 1), 2)) 
+
+search_query = {
+    "query": {
+        "knn": {
+            "values": {
+                "vector": vec, 
+                "k": 3
+            }
+        }
+    }
+}
+
+results = client.search(index=index_name, body=search_query)
```

Comment:
can we add that if we disable the _source field than it helps in reducing the latency of the search.

Reply:
we could, but feels like that's a very narrow use-case where I don't actually have any additional data and only want the ID, no?